server/images.go

```diff
@@ -1105,17 +1115,27 @@ func PullModel(ctx context.Context, name string, regOpts *RegistryOptions, fn fu
 	return nil
 }
 
-func pullModelManifest(ctx context.Context, mp ModelPath, regOpts *RegistryOptions) (*ManifestV2, error) {
+func pullModelManifest(ctx context.Context, mp ModelPath, currentDigest string, regOpts *RegistryOptions) (*ManifestV2, error) {
 	requestURL := mp.BaseURL().JoinPath("v2", mp.GetNamespaceRepository(), "manifests", mp.Tag)
 
 	headers := make(http.Header)
 	headers.Set("Accept", "application/vnd.docker.distribution.manifest.v2+json")
+
+	if currentDigest != "" {
+		headers.Set("If-None-Match", currentDigest)
+	}
+
 	resp, err := makeRequestWithRetry(ctx, http.MethodGet, requestURL, headers, nil, regOpts)
 	if err != nil {
 		return nil, err
 	}
 	defer resp.Body.Close()
 
+	// todo we can potentially read the manifest locally and return it here
+	if resp.StatusCode == http.StatusNotModified {
+		return nil, nil
```

Comment:
returning nil, nil can be error prone. instead it should either return the `current manifest, nil`, or `nil, os.ErrExist`. either one can be explicitly handled. 

returning the current manifest allows no changes in the upstream code which will print the progress as if it's pulling.

returning an error allows nicer handling for no updates
```go
manifest, err := pullModelManifest()
if errors.Is(err, os.ErrExist) {
  fn(api.ProgressResponse{Status: "model up-to-date"})
  return nil
} else if err != nil {
  return err
}
```